pages/categorization.vue

```diff
@@ -0,0 +1,521 @@
+<template>
+
+	<b-container fluid>
+
+		<tool-navbar :navItemsState="navItemsState" :pageName="fullName"></tool-navbar>
+
+		<b-row>
+
+			<b-col cols="4">
+				<coloring-listgroup
+					:columnData="recommendedColumns"
+					:defaultPalette="$store.state.pageData.categorization.default"
+					tag="recommended-column"
+					title="Recommended Columns"
+					instructions="Click column type and then corresponding column from tsv file"
+					v-on:paint-action="$store.dispatch('saveCurrentPaintInfo', $event)"
+					>
+				</coloring-listgroup>
+			</b-col>
+
+			<b-col cols="8">
+				<filedata-table
+					:fields="columnMatchTableFields"
+					:paintClass="getPaintClass"
+					:currentPalette="$store.state.pageData.categorization.current"
+					:defaultPalette="$store.state.pageData.categorization.default"
+					:tableData="tableDataFromTsvAndOrJson"
+					v-on:column-name-selected="tableClick($event)">
+				</filedata-table>
+			</b-col>
+
+		</b-row>
+
+		<b-row>
+			
+			<b-col cols="9"></b-col>
+			
+			<b-col cols="3">
+				<!-- Only enabled when at least one column has been categorized -->
+				<b-button
+					class="float-right"
+					:disabled="!readyForNextStepFlag"
+					:to="'/' + pageNames.annotation.location"
+					:variant="nextPageButtonColor">
+					Next step: Annotate columns
+				</b-button>
+			</b-col>
+			
+		</b-row>		
+
+	</b-container>
+
+</template>
+
+<script>
+
+	// String literals
+
+	export default {
+
+		name: "CategorizationPage",
+
+		created() {
+
+			// Determine page state from data contents and change to that new state
+			this.changeToNewState();
+		},		
+
+		data() {
+
+			return {
+
+				// Columns for file data table	
+				columnMatchTableFields: [
+
+					{ key: "column" },
+					{ key: "description" }
+				],
+
+				// Current state of the page
+				currentState: 0,
+
+				// Default table background color
+				defaultBackgroundColor: this.$store.getters.pageData.categorization.default.bColor,
+
+				// Full text name of this page
+				fullName: this.$store.getters.pageNames.categorization.fullName, 
+
+				// Status of the navbar item links for other pages
+				navItemsState: [
```

Comment:
I'd actually prefer the navbar not to have global state (and not just because we're not using any form of template inheritance). It's better to have the navbar state for a particular page be determined by the actual data in the store and not some state value we save for it. This way we don't have to worry about updating said saved state values in the store and deal with stale state bugs.

In the current implementation, conditionally checking the stored data each page requires for access to the successive step in the annotation workflow is sufficient enough to determine the visual state of the page (enabled buttons, enabled nav items, populated textboxes, etc.)